resources/views/faults/list.blade.php

```diff
@@ -0,0 +1,115 @@
+<div id="faults-table"></div>
+<script src="js/moment.min.js"></script>
+<script type="application/javascript">
+    $(document).ready(function () {
+        function status_translate(cell) {
+            if (typeof cell !== "string") {
+                cell = cell.getValue();
+            }
+            const translations = {
+                UNSEEN: "@lang('faults.unseen')",
+                SEEN: "@lang('faults.seen')",
+                DONE: "@lang('faults.done')",
+                WONT_FIX: "@lang('faults.wont_fix')"
+            };
+            const colors = {
+                UNSEEN: "text-secondary",
+                SEEN: "text-info",
+                DONE: "text-success",
+                WONT_FIX: "text-danger"
+            };
+            return '<div class="font-italic ' + colors[cell] + ' text-right">' + translations[cell] + '</div>';
+        }
+
+        var button = function (cell, formatterParams) {
+            switch (formatterParams['status']) {
+                case "DONE":
+                    var style = "btn-success";
+                    var text = "@lang('faults.done')";
+                    break;
+            
+                case "WONT_FIX":
+                    var style = "btn-danger";
+                    var text = "@lang('faults.wont_fix')";
+                    break;
+                
+                case "UNSEEN":
+                    var style = "btn-danger";
+                    var text = "@lang('faults.reopen')";
+                    break;
+            }
+
+            return $("<button type=\"button\" class=\"btn btn-sm " + style + " float-left\">" + text + "</button>").click(function () {
+                var data = cell.getRow().getData();
+                confirm('@lang('faults.confirm')', '@lang('faults.confirm_message')' + text, '@lang('faults.cancel')', '@lang('faults.confirm')', function() {
+                    $.post("{{ route('faults.update') }}", {id: data.id, status: formatterParams['status']}, function(data){
+                        if (data !== "true"){
+                            alert('@lang('faults.autherror')');
+                        } else {
+                            window.location.reload(true);
+                        }
+                    });
+                });
+            })[0];
+        };
+
+        var button_formatter = function (cell, formatterParams, onRendered) {
+            var status = cell.getRow().getData()["status"];
+
+            switch (formatterParams['status']) {
+                case "DONE":
```

Comment:
You could change the statuses here and below to the consts declared.

Reply:
OK, changed all of them to the constants.